agent/rpcclient/health/health.go

```diff
@@ -49,6 +50,7 @@ func (c *Client) getServiceNodes(
 ) (structs.IndexedCheckServiceNodes, cache.ResultMeta, error) {
 	var out structs.IndexedCheckServiceNodes
 
+	req.ServiceName = strings.ToLower(req.ServiceName)
```

Comment:
Is this line necessary with the `strings.EqualFold` change to filtering?

Is this because the cache key uses this value? If that is the case I guess we should be normalizing this strings in the function which creates the cache key instead of here.

Reply:
@dnephin This is not exactly mandatory (the only drawback would be having 2 separate entries when requesting /health/service/NAME vs /health/service/name), but works better as only 1 entry will be cached

The big advantage of doing it there is that we know we are handling a Service (with its case insensitive semantics), we could do it in:
```
(c *Cache) Get(ctx context.Context, t string, r Request)
```

But, we should in such case either:
* add a specific case for services
* add a new method to compute a key which is different for health than for kv values for instance (aka before calling  `key := makeEntryKey(r.TypeEntry.Name, r.Info.Datacenter, r.Info.Token, r.Info.Key)`

Those 2 solutions seem more complex to me and more error-prone. Do you have another idea?

Reply:
Oh ya, I was thinking it would be good to put it here (https://github.com/hashicorp/consul/blob/master/agent/structs/structs.go#L605) where we build the key for this cache type.

Reply:
@dnephin Done!

Reply:
Looks good! I guess we should also remove this line now as well, since it is done in the cache-type?